### PR TITLE
Change deprecation to error for C-style array syntax

### PR DESCRIPTION
**Pull request by JinShil**
Documentation update:  https://github.com/dlang/dlang.org/pull/2401

### CODE REVIEW

File: src/dmd/parse.d [hunk 1]
```diff
@@ -5030,10 +5030,8 @@ final class Parser(AST) : Lexer
 
         const(char)* sp = !ident ? "" : " ";
         const(char)* s = !ident ? "" : ident.toChars();
-        if (alt & 1) // contains C-style function pointer syntax
-            error(loc, "instead of C-style syntax, use D-style `%s%s%s`", t.toChars(), sp, s);
-        else
-           dmd.errors.deprecation(loc, "instead of C-style syntax, use D-style syntax `%s%s%s`", t.toChars(), sp, s);
+        // @@@DEPRECATED_2.087_REMOVE@@@
```
Comment (wilzbach) [inline src/dmd/parse.d]: Do we plan to ever deprecate this?
It seems like a useful error message to keep around forever (it was very helpful for the C->D conversion).

**Review by wilzbach — APPROVED** [review]

File: src/dmd/parse.d (hunk 1 above)
Comment (JinShil) [reply src/dmd/parse.d]: Actually, I hope so.  I don't think we should spend logic detecting invalid C syntax just so we can display an error.  I'm hoping we can just rely on ordinary parsing errors.  But that's just my opinion.  I'll go along with whatever the team decides.

Comment (Geod24) [reply src/dmd/parse.d]: I used to think the same as you @JinShil then I ported some C and C++ code to D. If the overhead isn't too much (it's usually a few lines), it's definitely worth it.

**Review by jacob-carlborg — APPROVED** [review]

**Comment by wilzbach** [comment]
Fails due to:

```
Test the D Language specification
../dmd/generated/linux/release/64/dmd -run tools/dspec_tester.d --compiler=../dmd/generated/linux/release/64/dmd
---
void main() {
import std.stdio;
enum Color { red, blue, green };

int value[Color.max + 1] =
  [ Color.blue :6,
    Color.green:2,
    Color.red  :5 ];

}
---
__stdin.d(3): Deprecation: use `{ }` for an empty statement, not `;`
__stdin.d(5): Error: instead of C-style syntax, use D-style `int[Color.max + 1] value`
```

PR: https://github.com/dlang/dlang.org/pull/2402

**Pull request merged**